### PR TITLE
Test runner orchestrator

### PR DESCRIPTION
**Pull request by nicojs**
Added the test runner orchestrator, responsible for managing test runners. This still needs to change because i just added 3 properties to Mutant because i needed them:
```typescript
  public timeSpentScopedTests: number;
  public scopedTestIds: number[];
  public specsRan: string[];
```
We will need to fill these properties or provide other ways of reaching them. That should not be the responsibility of the orchestrator, that's why i didn't include it in this PR

### CODE REVIEW

File: src/TestRunnerOrchestrator.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        runnerAdapter: IsolatedTestRunnerAdapterFactory.create(this.createTestRunSettings(tempSourceFiles, selector, this.options.port + portOffset, false)),
+        selector
+      };
+    });
+  }
+
+  private copyAllSourceFilesToTempFolder() {
+    return new Promise<FileMap>((resolve, reject) => {
+      let fileMap: FileMap = Object.create(null);
+      var tempFolder = StrykerTempFolder.createRandomFolder('test-runner-source-files');
```
Comment (simondel) [inline src/TestRunnerOrchestrator.ts]: Who removes this folder and when?

Comment (nicojs) [inline src/TestRunnerOrchestrator.ts]: Well... i think at the end we can simply remove the entire temp folder. For Milestone 1 (and the beta) i think we do not need to worry about it. The temp folder is created in your temp directory, people are pretty used to cleaning the temp directory from time to time. I see karma is also not cleaning the temp directory.. so maybe don't worry about it? :)

Comment (nicojs) [inline src/TestRunnerOrchestrator.ts]: Also: I still have my doubts about the `StrykerTempFolder` being in the api folder and about the name. We should discuss this before V1

**Comment by nicojs** [comment]
I just added a commit for the removal of the temp mutation api. It was still in there even though we discussed to not implement it in v1. Sorry :)

File: src/TestRunnerOrchestrator.ts (hunk 1 above)
Comment (simondel) [inline src/TestRunnerOrchestrator.ts]: I agree that we should discuss this before V1 :)

**Comment by simondel** [comment]
It looks good, well done!

**Pull request merged**